### PR TITLE
test: design testing strategy

### LINKED ISSUE #98: Design testing strategy
**Is your feature request related to a problem? Please describe.**

Before publishing, we need to see how the tool performs against libraries for which the metadata is known. A collection for libraries for which (some of) the metadata is already mined is [here](https://docs.google.com/spreadsheets/d/1r4QolWT3uFnmAVTyGrcNnVEy78-4j446-S44jogxx5I/edit). But as this list will be expanding, we need to make sure that we have a testing strategy in place to easily run these tests.

**Describe the solution you'd like**
The testing strategy should include
- Fetching of the data ([this](https://github.com/zavolanlab/zarp/blob/dev/workflow/rules/sra_download.smk) Snakemake workflow can be used)
- Arranging of the data in some defined directory structure
- Convert SRA to FASTQ (`fastqdump` from SRA toolkit, there may be an alternative `fasterqdump`)
- Running HTSinfer over all files in the directory
- Compilation of the results in a matrix format
- Comparing the results with the mined data, as collected in the metadata sheet linked above
- Analysis of comparison results
- Reporting of final test results

### LINKED ISSUE #110: Run first tests

### PR DESCRIPTION
**Pull request by balajtimate**
### Description

Testing strategy for running HTSinfer with parameters and generating plots and table of results.
First tests were run using this script.
Dependecies:
`seaborn v0.12.2`

Resolves #98 
Resolves #110

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- [ ] Documentation update

### Checklist

Please carefully read these items and tick them off if the statements are true
_or do not apply_.

- [X] I have performed a self-review of my own code
- [X] My code follows the existing coding style, lints and generates no new
      warnings
- [X] I have added type annotations to all function/method signatures, and I
      have added type annotations for any local variables that are non-trivial,
      potentially ambiguous or might otherwise benefit from explicit typing.
- [X] I have commented my code in hard-to-understand areas
- [X] I have added ["Google-style docstrings"] to all new modules, classes,
      methods/functions or updated previously existing ones
- [X] I have added tests that prove my fix is effective or that my feature
      works
- [X] New and existing unit tests pass locally with my changes and I have not
      reduced the code coverage relative to the previous state
- [X] I have updated any sections of the app's documentation that are affected
      by the proposed changes

If for some reason you are unable to tick off all boxes, please leave a
comment explaining the issue you are facing so that we can work on it
together.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -115,3 +115,7 @@ results
 .DS_Store
 tests/.DS_Store
 results_htsinfer
+tests/cluster_tests/.snakemake
```
Comment (uniqueg) [inline .gitignore]: I think you could just put `.snakemake` - we don't want this folder under version control under any circumstance, regardless of where it is created.

File: .gitignore [hunk 2]
```diff
@@ -115,3 +115,7 @@ results
...
+tests/cluster_tests/zarp
+tests/cluster_tests/results_sra_downloads
+tests/cluster_tests/mined_test_data_result.tsv
```
Comment (uniqueg) [inline .gitignore]: I think we may actually include this in the repo, no? Won't hurt

File: .gitignore [hunk 3]
```diff
@@ -115,3 +115,7 @@ results
 .DS_Store
 tests/.DS_Store
 results_htsinfer
+tests/cluster_tests/.snakemake
+tests/cluster_tests/zarp
```
Comment (uniqueg) [inline .gitignore]: What is this? This might be better suited in `.git/exclude` in your personal copy of the repo. I don't see why anyone else would consider putting a `zarp` folder here?

File: data/adapter_fragments.txt [hunk 4]
```diff
@@ -21,3 +21,4 @@ TCGTATGCCGTCTTC
 TCGTCGGCAGCGTCA
 TGGAATTCTCGGGTG
 AGATCGGAAGAGCGT
+AGATCGGAAGAGCAC
```
Comment (uniqueg) [inline data/adapter_fragments.txt]: This should go in a separate PR - it has nothing to do with the test strategy.

File: htsinfer/get_read_layout.py [hunk 5]
```diff
@@ -20,7 +20,8 @@
 
 
 class GetReadLayout:
-    """Determine the adapter sequence present in the FASTQ sequencing libraries.
+    """Determine the adapter sequence present in the FASTQ sequencing
+    libraries.
```
Comment (uniqueg) [inline htsinfer/get_read_layout.py]: This also doesn't have anything to do with the test strategy.

**Review by uniqueg — CHANGES_REQUESTED** [review]

File: data/adapter_fragments.txt (hunk 4 above)
Comment (balajtimate) [reply data/adapter_fragments.txt]: You're right, sorry, I deleted it and will put it into a separate PR with the other ones I found.

File: .gitignore (hunk 3 above)
Comment (balajtimate) [reply .gitignore]: It was left here from a previous commit when I had `zarp` there, it's fixed now.

File: htsinfer/get_read_layout.py (hunk 5 above)
Comment (balajtimate) [reply htsinfer/get_read_layout.py]: I created another PR for this if that's okay.

**Review by uniqueg — APPROVED** [review]

**Review by uniqueg — CHANGES_REQUESTED** [review]
Please fix the failing checks, otherwise looks good to me

**Comment by balajtimate** [comment]
The checks now all passed, the requested changes are made.

**Review by uniqueg — APPROVED** [review]
